pkg/apis/navigator/validation/elasticsearch_test.go

```diff
@@ -0,0 +1,379 @@
+			NodeSelector: validNodePoolNodeSelector,
+			Resources:    validNodePoolResources,
+			Persistence:  validNodePoolPersistenceConfig,
+		},
+		"invalid roles": {
+			Name:         validNodePoolName,
+			Replicas:     validNodePoolReplicas,
+			Roles:        invalidRoles,
+			NodeSelector: validNodePoolNodeSelector,
+			Resources:    validNodePoolResources,
+			Persistence:  validNodePoolPersistenceConfig,
+		},
+		"missing roles": {
+			Name:         validNodePoolName,
+			Replicas:     validNodePoolReplicas,
+			NodeSelector: validNodePoolNodeSelector,
+			Resources:    validNodePoolResources,
+			Persistence:  validNodePoolPersistenceConfig,
+		},
+		"invalid persistence config": {
+			Name:         validNodePoolName,
+			Replicas:     validNodePoolReplicas,
+			Roles:        validNodePoolRoles,
+			NodeSelector: validNodePoolNodeSelector,
+			Resources:    validNodePoolResources,
+			Persistence:  invalidPersistenceConfig,
+		},
+	}
+
+	successCases := []navigator.ElasticsearchClusterNodePool{
+		{
+			Name:         validNodePoolName,
+			Replicas:     validNodePoolReplicas,
+			Roles:        validNodePoolRoles,
+			NodeSelector: validNodePoolNodeSelector,
+			Resources:    validNodePoolResources,
+			Persistence:  validNodePoolPersistenceConfig,
+		},
+		{
+			Name:  validNodePoolName,
+			Roles: validNodePoolRoles,
+		},
+	}
+
+	for i, successCase := range successCases {
+		t.Run("success case "+strconv.Itoa(i), func(t *testing.T) {
+			if errs := validation.ValidateElasticsearchClusterNodePool(&successCase, field.NewPath("test")); len(errs) != 0 {
+				t.Errorf("expected success: %v", errs)
+			}
+		})
+	}
+
+	for n, test := range errorCases {
+		t.Run(n, func(t *testing.T) {
+			errs := validation.ValidateElasticsearchClusterNodePool(&test, field.NewPath("test"))
+			if len(errs) == 0 {
+				t.Errorf("Expected errors to be returned for spec %+v but got none", test)
+			}
+			for _, err := range errs {
+				field := err.Field
+				if !strings.HasPrefix(field, "test") &&
+					field != "test.name" &&
+					field != "test.replicas" &&
+					field != "test.roles" &&
+					field != "test.nodeSelector" &&
+					field != "test.resources" &&
+					field != "test.persistence" {
+					t.Errorf("%s: missing prefix for: %v", n, err)
+				}
+			}
+		})
+	}
+}
+
+func TestValidateElasticsearchPersistence(t *testing.T) {
+	validSize := resource.MustParse("10Gi")
+	errorCases := map[string]navigator.ElasticsearchClusterPersistenceConfig{
+		"enabled but no size specified": navigator.ElasticsearchClusterPersistenceConfig{
+			Enabled: true,
+		},
+		"enabled but invalid size specified": navigator.ElasticsearchClusterPersistenceConfig{
+			Enabled: true,
+			Size:    *resource.NewQuantity(-1, resource.BinarySI),
+		},
+	}
+	successCases := []navigator.ElasticsearchClusterPersistenceConfig{
+		{},
+		{
+			Size: validSize,
+		},
+		{
+			StorageClass: "something",
+		},
+		{
+			Enabled: true,
+			Size:    validSize,
+		},
+	}
```

Comment:
Thanks for these tests.
I think that the size should be validated and be required to be > 0, regardless of whether persistence is enabled.

Reply:
So if persistence is enabled, size MUST be >0 (cannot be zero).
If persistence is *disabled*, we should accept the zero value for size, *or* an otherwise valid value (despite it being ignored) but we should NEVER allow a negative value. I'll update/ensure our tests cover these cases.